ignite/metrics/confusion_matrix.py

```diff
@@ -238,3 +238,27 @@ def cmRecall(cm, average=True):
     if average:
         return recall.mean()
     return recall
+
+
+def DiceCoefficient(cm, ignore_index=None):
+    """Calculates Dice Coefficient for a given Confusion Matrix.
+
+    Args:
+        cm (ConfusionMatrix): instance of confusion matrix metric
+        ignore_index (int, optional): index to ignore, e.g. background index
+    """
+
+    if not isinstance(cm, ConfusionMatrix):
+        raise TypeError("Argument cm should be instance of ConfusionMatrix, but given {}".format(type(cm)))
+
+    if ignore_index is not None:
+        if not (isinstance(ignore_index, numbers.Integral) and 0 <= ignore_index < cm.num_classes):
+            raise ValueError("ignore_index should be non-negative integer, but given {}".format(ignore_index))
+        cm.confusion_matrix[ignore_index] = 0
```

Comment:
I think this is a not good idea to modify `confusion_matrix` as it can be, for example, also used in other metrics which use this info.